lib/src/main/java/uk/gov/di/ipv/cri/passport/library/service/ClientFactoryService.java

```diff
@@ -76,91 +71,30 @@ public SqsClient getSqsClient() {
                 .build();
     }
 
+    // ThreadLocalRandom not used cryptographically here
+    @java.lang.SuppressWarnings("java:S2245")
     public SSMProvider getSSMProvider() {
         SsmClient ssmClient =
                 SsmClient.builder()
                         .region(awsRegion)
                         .httpClient(UrlConnectionHttpClient.create())
                         .build();
 
-        return ParamManager.getSsmProvider(ssmClient).defaultMaxAge(300, ChronoUnit.SECONDS);
-    }
+        // A random cache age between 5-15 minutes (in seconds)
+        // Avoids multiple scaling lambdas expiring their caches at the exact same time
+        int maxCacheAge = ThreadLocalRandom.current().nextInt(900 - 300 + 1) + 300;
```

Comment:
Have we checked this cache is reset by redeploying

Reply:
For all the current lambdas which are using the parameter store service - Yes. This will be done in the handlers constructors via the calls to getParameterStoreService() which in-turn calls getSSMProvider() setting the cache for that lambda context.